### PR TITLE
[DATASET] Add cnn_dailymail dataset

### PR DESCRIPTION
**Pull request by gongel**
### PR types

New features

### PR changes

Others

### Description

- Add cnn_dailymail dataset
- Use like this: 
```python
from paddlenlp.datasets import load_dataset
train_set = load_dataset("cnn_dailymail",  splits=["train"]) # version defaults to "3.0.0"
train_set, dev_set, test_set = load_dataset("cnn_dailymail",  splits=["train", "dev", "test"], version="3.0.0")
```

### CODE REVIEW

File: paddlenlp/datasets/cnn_dailymail.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+                next_is_highlight = True
+            elif next_is_highlight:
+                highlights.append(line)
+            else:
+                article_lines.append(line)
+
+        # Make article into a single string
+        article = " ".join(article_lines)
+
+        if tfds_version >= "2.0.0":
```
Comment (ZeyuChen) [inline paddlenlp/datasets/cnn_dailymail.py]: Don't envolve tfds_version concept to paddlenlp dataset, is it possible just use the latest version?

**Review by ZeyuChen — COMMENTED** [review]
Is it possible just use the latest 3.0.0 version instead of involve tfds concept?

File: paddlenlp/datasets/cnn_dailymail.py (hunk 1 above)
Comment (gongel) [reply paddlenlp/datasets/cnn_dailymail.py]: I think paddlenlp should be consistent with the original data. For ease of use, ` version` defaults to "3.0.0".

**Review by smallv0221 — CHANGES_REQUESTED** [review]
我测试一下没读到数据，你测试过了么

File: paddlenlp/datasets/cnn_dailymail.py (hunk 1 above)
Comment (smallv0221) [reply paddlenlp/datasets/cnn_dailymail.py]: 这里还是用version吧

**Comment by gongel** [comment]
> 我测试一下没读到数据，你测试过了么

我这边测试没问题，第一次加载解压比较慢。

File: paddlenlp/datasets/cnn_dailymail.py (hunk 1 above)
Comment (gongel) [reply paddlenlp/datasets/cnn_dailymail.py]: > 这里还是用version吧

Done, thx

**Review by smallv0221 — APPROVED** [review]
LGTM

**Pull request merged**